src/main/java/com/splunk/hecclient/HecAckPoller.java

```diff
@@ -252,14 +297,16 @@ private void findAndRemoveTimedoutBatches(Map<Long, EventBatch> batches, List<Ev
 
     private void handleAckPollResponse(String resp, HecChannel channel) {
         log.debug("ackPollResponse={}, channel={}", resp, channel);
-        HecAckPollResponse ackPollResult;
+        HecAckPollResponse ackPollResult = null;
         try {
             ackPollResult = jsonMapper.readValue(resp, HecAckPollResponse.class);
         } catch (Exception ex) {
             log.error("failed to handle ack polled result", ex);
             return;
         }
-        handleAckPollResult(channel, ackPollResult);
+        if (ackPollResult != null) {
```

Comment:
look for more graceful way of handling null response. use exception handling or remove if not needed